{
  long[] x={0};
  boolean thrown=false;
  try {
    Array.setLong(x,0,1);
  }
 catch (  Exception e) {
    fail("Exception during get test : " + e.getMessage());
  }
  assertEquals("Get returned incorrect value",1,Array.getLong(x,0));
  try {
    Array.setLong(new Object(),0,9);
  }
 catch (  IllegalArgumentException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Passing non-array failed to throw exception");
  }
  thrown=false;
  try {
    Array.setLong(x,4,9);
  }
 catch (  ArrayIndexOutOfBoundsException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Invalid index failed to throw exception");
  }
  thrown=false;
  try {
    Array.setLong(null,0,0);
  }
 catch (  NullPointerException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Null argument failed to throw NPE");
  }
}
